algorithms/linfa-clustering/src/appx_dbscan/cells_grid/cell.rs

```diff
@@ -45,58 +41,41 @@ pub struct CoreCellInfo<F: Float> {
     i_cluster: usize,
 }
 
-impl<F: Float> CoreCellInfo<F> {
-    fn new() -> CoreCellInfo<F> {
-        CoreCellInfo {
-            root: TreeStructure::new_empty(),
-            i_cluster: 0,
-        }
-    }
-}
-
 #[derive(Clone)]
 /// A cell from a grid that partitions the D dimensional euclidean space.
 pub struct Cell<F: Float> {
     /// The index of the intervals of the D dimensional axes where this cell lies
-    index: Array1<i64>,
+    pub index: Array1<F>,
```

Comment:
why is the index a floating point?

Reply:
Nearest Neighbours algorithms only work with floats, so having float indices prevents us from having to convert the indices into floats when using `linfa-nn`.